### PR TITLE
Add pull request closed event

### LINKED ISSUE #286: Support PR closed event
To support something like review environments it would be nice to be able to listen on PR closed events. This would allow to start and update a review environment in the moment a PR is created or someone pushed to it and destroy it after the PR has been closed or merged.

### PR DESCRIPTION
**Pull request by anbraten**
- [x] updates docs
- [x] adjust UI
  - [x] show correct icon
  - [x] show correct link (to pr)
  - [x] add as option in secret edit
- [x] parse webhook
- [x] update tests
  - [x] github merged
  - [x] github closed
  - [x] gitea merged
  - [x] gitea closed
  - [x] bitbucket merged
  - [x] bitbucket closed
  - [x] gitlab merged
  - [x] gitlab closed

closes #286

### CODE REVIEW

File: server/forge/bitbucket/parse.go [hunk 1]
```diff
@@ -71,14 +73,14 @@ func parsePushHook(payload []byte) (*model.Repo, *model.Pipeline, error) {
...
-// details. If the pull request is closed nil values are returned.
+// details.
 func parsePullHook(payload []byte) (*model.Repo, *model.Pipeline, error) {
 	hook := internal.PullRequestHook{}
 
 	if err := json.Unmarshal(payload, &hook); err != nil {
 		return nil, nil, err
 	}
-	if hook.PullRequest.State != stateOpen {
+	if hook.PullRequest.State != stateOpen && hook.PullRequest.State != stateClosed {
```
[the remark this answers is not in the document]
Comment (anbraten) [reply server/forge/bitbucket/parse.go:83]: The plan is that this event is triggered for both as the use-case which I heard most of the time / want this by myself is closing sth like review environments as soon as a PR got closed / merged, no matter the reason.

[the remark this answers is not in the document]
Comment (anbraten) [reply server/forge/bitbucket/parse.go:83]: Testing a forges is still left and kina difficult as the docs for webhooks are just bad for all 4 of them 😕

File: docs/docs/20-usage/50-environment.md [hunk 2]
```diff
@@ -46,93 +46,93 @@ steps:
...
+| `CI_PREV_COMMIT_MESSAGE`         | previous commit message                                                                         |
+| `CI_PREV_COMMIT_AUTHOR`          | previous commit author username                                                                 |
+| `CI_PREV_COMMIT_AUTHOR_EMAIL`    | previous commit author email address                                                            |
+| `CI_PREV_COMMIT_AUTHOR_AVATAR`   | previous commit author avatar                                                                   |
+|                                  | **Previous pipeline**                                                                           |
+| `CI_PREV_PIPELINE_NUMBER`        | previous pipeline number                                                                        |
+| `CI_PREV_PIPELINE_PARENT`        | previous pipeline number of parent pipeline                                                     |
+| `CI_PREV_PIPELINE_EVENT`         | previous pipeline event (push, pull_request, pull_request_closed, tag, deployment, manual)      |
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/docs/20-usage/50-environment.md]: moved the list to its own section

File: web/src/components/repo/pipeline/PipelineItem.vue [hunk 3]
```diff
@@ -37,7 +37,10 @@
         class="grid grid-rows-2 grid-flow-col w-full md:ml-auto md:w-96 py-2 gap-x-4 gap-y-2 flex-shrink-0 text-wp-text-100"
       >
         <div class="flex space-x-2 items-center min-w-0">
-          <Icon v-if="pipeline.event === 'pull_request'" name="pull_request" />
+          <Icon
+            v-if="pipeline.event === 'pull_request' || pipeline.event === 'pull_request_closed'"
+            name="pull_request"
+          />
```
[the remark this answers is not in the document]
Comment (anbraten) [reply web/src/components/repo/pipeline/PipelineItem.vue]: done

File: server/forge/bitbucket/parse.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anbraten) [reply server/forge/bitbucket/parse.go:83]: got webhook fixtures for all forges now

**Comment by 6543** [comment]
@anbraten ci fail related

**Comment by anbraten** [comment]
I have to update the tests.

**Comment by 6543** [comment]
do you have time to do so? or should we move to nex release?

File: web/src/components/repo/pipeline/PipelineStepList.vue [hunk 4]
```diff
@@ -11,11 +11,11 @@
         <span>{{ pipeline.author }}</span>
       </div>
       <a
-        v-if="pipeline.event === 'pull_request'"
+        v-if="pipeline.event === 'pull_request' || pipeline.event === 'pull_request_closed'"
         class="flex items-center space-x-1 text-wp-link-100 hover:text-wp-link-200 min-w-0"
         :href="pipeline.forge_url"
       >
-        <Icon name="pull_request" />
+        <Icon name="pull-request" />
```
Comment (6543) [inline web/src/components/repo/pipeline/PipelineStepList.vue:18]: intentional ?!?

**Review by 6543 — APPROVED** [review]

File: web/src/components/repo/pipeline/PipelineStepList.vue (hunk 4 above)
Comment (anbraten) [reply web/src/components/repo/pipeline/PipelineStepList.vue:18]: Yes to be consistent

**Pull request merged**